src/beamlime/applications/_nexus_helpers.py

```diff
@@ -1,155 +1,365 @@
+
+
+DataModuleStore = dict[NexusPath, DataModuleParent]
+
+
+def create_dataset(
+    *, name: str, dtype: str, initial_values: Any, unit: str | None = None
+) -> _NexusDataset:
+    """Creates a dataset according to the arguments."""
+    dataset: _NexusDataset = {
+        "module": "dataset",
+        "config": {
+            "name": name,
+            "dtype": dtype,
+            "values": initial_values,
+        },
+        "attributes": [],
+    }
+    if unit is not None:
+        attrs = dataset["attributes"]
+        attrs.append(
+            {
+                "name": "units",
+                "dtype": "string",
+                "values": unit,
+            }
         )
-        group['children'].append(dataset)
     return dataset
 
 
-def _merge_ev44(group, message):
-    for field, value in message.items():
-        if value is None:  # Could be monitor without pixel_id
-            continue
-        dataset = _init_dataset(group, field, value)
-        if field in ('time_of_flight', 'pixel_id'):
-            dataset['config']['values'] = np.concatenate(
-                (dataset['config']['values'], value)
-            )
+ModuleNameType = Literal["ev44"]  # "f144", "tdct" will be added in the future
+"""Name of the module that is supported by beamlime."""
+
+
+def _get_array_values(dataset: Mapping) -> np.ndarray:
+    return dataset["config"]["values"]
+
+
+def _set_values(dataset: Mapping, values: Any) -> None:
+    dataset["config"]["values"] = values
+
 
+def _initialize_ev44(group: DataModuleParent) -> None:
+    """Initialize ev44 datasets in the parent.
 
-def _merge_f144(group, message):
-    for field, value in message.items():
-        dataset = _init_dataset(group, field, value)
-        if field in ('timestamp', 'value'):
-            dataset['config']['values'] = np.stack((dataset['config']['values'], value))
+    Params
+    ------
+    group:
+        A data group that has a module place holder as a child.
 
+    Side Effects
+    ------------
+    ``children`` of the ``group`` will have 4 datasets with empty values.
 
-def _merge_tdct(group, message):
-    for field, value in message.items():
-        dataset = _init_dataset(group, field, value)
-        if field in ('timestamps',):
-            dataset['config']['values'] = np.concatenate(
-                (dataset['config']['values'], value)
+    - event_time_zero
+    - event_time_offset
+    - event_index
+    - event_id
+
+    This function doesn't remove a place holder from the ``children``.
+    It is expected to be removed outside of this function.
+
+    """
+    children = group.setdefault("children", [])
+    # event_time_zero
+    children.append(
+        create_dataset(
+            name="event_time_zero",
+            dtype="int",
+            initial_values=np.asarray([]),
+            unit="ns",
+        )
+    )
+    # event_time_offset
+    children.append(
+        create_dataset(
+            name="event_time_offset",
+            dtype="int",
```

Comment:
we should specify the right `dtype` here as specified in `nexus-streaming-types`, that is, set `int32` or `int64`.

Reply:
Hmm... yeah. 
Does it mean that we use `int64` if they are `long` and `int32` if they are just `int` in the schema?
For example:

https://github.com/ess-dmsc/streaming-data-types/blob/13dd32d1aa7f8eb47cbb1393918e6929f9890f17/schemas/ev44_events.fbs#L5C1-L20C2

Reply:
Yes I think that makes sense.